### ISSUE TITLE
Breadcrumb for individual court records

### ISSUE THREAD

**Issue by robertocarroll** [robertocarroll/icc-beta#12]
https://github.com/robertocarroll/icc-alpha/issues/88

More context for a particular court record in the breadcrumb.

Home › Democratic Republic of the Congo › The Prosecutor v. Thomas Lubanga Dyilo

Home > filing

**Comment by robertocarroll** [comment]
Can we use css to hide the default breadcrumb on the court record template and then use javascript to create a custom breadcrumb (looks exactly the same and same position as the default breadcrumb) consisting of information which you are already passing to the url, e.g. situation, case, perhaps even source?

**Comment by kyleschaeffer** [comment]
This will be very difficult to do because even if you generate a breadcrumb from the URL (which is feasible), there is no guarantee that that information will always be available.  Users can navigate to a page from search results, other web parts, links created on other pages, etc.  I don't find it feasible to change the page URL in all of these cases.

It may be easier to consider doing this server-side.  If we're already storing the metadata for situations and regions, perhaps we could generate a breadcrumb from this existing metadata?

**Comment by robertocarroll** [comment]
Thanks for the info. Just to clarify, I only meant this approach for court records - where the Sharepoint breadcrumb doesn't work and where the page is generated dynamically (to my understanding) from passing various metadata values to a single template.

**Comment by kyleschaeffer** [comment]
Understood, if we know the entry point is static, and that the URL will always be generated, that definitely makes things easier.

**Comment by simmigaba** [comment]
If its needed for Court Record pages, I think it should be feasible. I will give it a try and let you know.

**Comment by robertocarroll** [comment]
The latest design deals with this issue: the breadcrumb would anchor the search, sort, filter on the type, e.g. court record, news, publication, etc. and the top area shows the filter state and updates in response to the user action.
- Suggest it builds up into a readable sentence, e.g. 
  26 [Document type] in [language] for [case/situation] by [source] from [phase] x . 
- Will there be a single cross which will act like a clear all / reset filter?

![icc-court-records jpg](https://cloud.githubusercontent.com/assets/1565953/5280857/c3fcf810-7aee-11e4-91b6-f3ac4e28ee9c.jpg)

**Comment by kyleschaeffer** [comment]
I think each component of the search (like [Document type]) should include an "x" next to it. This will allow for the user to clear specific pieces of the search for more control. The "clear all" could be easier to use in some cases, but in others it might prove frustrating because users only wanted to clear one aspect of the search filters.

What do you think?

**Comment by robertocarroll** [comment]
Sounds good - as long as it's not too fiddly, i.e. the hit area is ample enough for each cross.

**Comment by simmigaba** [comment]
I do not understand how the latest design solves the issue originally mentioned. The specific Court Record Page e.g. http://sharepoint13vm.cloudapp.net/sites/icc/Pages/filing.aspx?docNo=ICC-01/04-01/06-3108-Corr will not always be re-directed by clicking the Title link in Court Records Index page. The control can come from anywhere in website. E.g. user can click Court Record Title link in Court Records web part on Case page -- http://sharepoint13vm.cloudapp.net/sites/icc/drc/lubanga which will then re-direct the control to generic filing page where the complete Court Record page is generated dynamically based on Court Record "Full Document Number".
Are you suggesting that we also generate the labels (i.e. readable sentence) on specific "single" Court Record page at top based on Court records metadata values? If Yes then how it helps as these values are already displayed clearly as part of Court Record and there is no further filtering to be done. Please refer to link:  http://sharepoint13vm.cloudapp.net/sites/icc/Pages/filing.aspx?docNo=ICC-01/04-01/06-3108-Corr

![image](https://cloud.githubusercontent.com/assets/5706403/5315179/39db4c32-7ca0-11e4-97a8-47461992270b.png)

**Comment by robertocarroll** [comment]
You're right, Simmi. Sorry for the mix-up. The breadcrumb issue on the filter is different from the breadcrumb on an individual court record and i conflated the two. 

A breadcrumb would still be desirable in my opinion on an individual court record.

**Comment by simmigaba** [comment]
Hi Rob. Thanks. Yes I will explore the option of generating the breadcrumb at server side while building the dynamic Court Record page. At the time of displaying the individual Court Record, I do have Case and Situation value and site URLs info so a custom breadcrumb to navigate back to Case or Situation page may be possible. But as also mentioned by Kyle in above comments there are still many places on website from where a control can be re-directed to individual Court Record page so we need to see how easy it will be to generate breadcrumb for all possible navigation scenarios. In some cases, the user can directly open the Court Record link as we will be distributing these around e.g. user can directly type http://sharepoint13vm.cloudapp.net/sites/icc/Pages/filing.aspx?docNo=ICC-01/04-01/06-3108-Corr in the web browser so in this case there won't be any breadcrumb but just Home > filing.
This requirement though desirable is lil bit complex to implement. Please let me know if you or Kyle think there is any other possible way to provide navigation context on individual Court Record page and I will be happy to explore the alternate option.
Thanks.

**Comment by simmigaba** [comment]
basically what I am trying to say is that we can always generate a custom breadcrumb navigation as Home > Situation > Case > filing 
without considering how the individual court record page is re-directed as we will always have Case and Situation value and Page URLs while generating individual Court record page. Please let me know what you think about this?

**Comment by robertocarroll** [comment]
I think that's better than Home > filing as it is at the moment. It gives the user some important orientation: Situation and case.

If that's straightforward and robust, then go with that ... and then we can see how it feels and how users respond. Does that sound OK?

**Comment by kyleschaeffer** [comment]
I agree, I think we should choose one breadcrumb structure and just go with that, it will be too difficult to determine the breadcrumb based on user behavior, which will always be changing from user to user.
